Data/Vector/Generic/Mutable.hs

```diff
@@ -774,6 +810,89 @@ unsafeExchange v i x = UNSAFE_CHECK(checkIndex) "unsafeExchange" i (length v)
                          unsafeWrite v i x
                          return y
 
+-- Folds
+-- -----
+
+forI_ :: (Monad m, MVector v a) => v (PrimState m) a -> (Int -> m b) -> m ()
```

Comment:
Shouldn't it be `PrimMonad m`?

Reply:
Not really. This function doesn't use anything from PrimMonad, Callback that passed as parameter usually does

Reply:
Isn't `PrimMonad m` required to use `PrimState m`? `PrimState` is defined inside the `PrimMonad` class, so I thought that constraint would be required to be able to use it.

Reply:
Sorry for late answer. Not quite. It's true that in order to define `PrimState` type family instance one have to define `PrimMonad` instance. But when it comes to use associated type work just like stand alone type families. Since we don't use any methods of `PrimMonad` we don't need to bring its dictionary here.